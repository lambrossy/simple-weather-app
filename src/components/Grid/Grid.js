import styled from "styled-components";

const Container = styled.div({
  display: "flex",
  flexWrap: "wrap",
  justifyContent: "center",
});

const Cell = styled.div({
  margin: 10,
});

const Grid = ({ children }) => (
  <Container>
    {children.map((cell) => (
      <Cell key={cell.key}>{cell}</Cell>
    ))}
  </Container>
);

export default Grid;
