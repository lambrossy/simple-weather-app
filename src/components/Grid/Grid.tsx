import { FC } from "react";
import styled from "styled-components";

const Container = styled.div({
  display: "flex",
  flexWrap: "wrap",
  justifyContent: "center",
});

const Cell = styled.div({
  margin: 10,
});

interface Props {
  children: JSX.Element[];
}

const Grid: FC<Props> = ({ children }) => (
  <Container>
    {children.map((cell: JSX.Element) => (
      <Cell key={cell.key}>{cell}</Cell>
    ))}
  </Container>
);

export default Grid;
