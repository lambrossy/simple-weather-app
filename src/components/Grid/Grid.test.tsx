import { render } from "@testing-library/react";
import Grid from "./Grid";

it("renders correctly", () => {
  const element = render(
    <Grid>{[<span key="1">hello</span>, <span key="2">world</span>]}</Grid>
  );
  expect(element.container).toMatchSnapshot();
});
