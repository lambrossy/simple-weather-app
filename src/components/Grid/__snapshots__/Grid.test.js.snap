// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sc-bdnxRM fwJTUD"
  >
    <div
      class="sc-gtsrHT jnDyyg"
    >
      <span>
        hello
      </span>
    </div>
    <div
      class="sc-gtsrHT jnDyyg"
    >
      <span>
        world
      </span>
    </div>
  </div>
</div>
`;
